sale_triple_discount/models/sale_order_line.py

```diff
@@ -131,3 +131,36 @@ def _prepare_invoice_line(self, **kwargs):
             }
         )
         return res
+
+    @api.depends("discount1", "discount2", "discount3")
+    def _compute_discount(self):
+        for line in self:
+            if line.check_compute_discounts():
+                return super()._compute_discount()
+            else:
+                discount_factor = 1.0
+                for discount in [line.discount1, line.discount2, line.discount3]:
+                    discount_factor *= (100.0 - discount) / 100.0
+                line.discount = 100.0 - (discount_factor * 100.0)
+
+    @api.depends("discount")
+    def _compute_discounts(self):
+        for line in self:
+            if line.check_compute_discounts():
+                line.discount1 = line.discount
```

Comment:
@matteonext You can't do that !  These fields are not computed. So, this is not syntaxically correct.